### PR TITLE
[RFC] vim-patch:7.4.1223

### PR DESCRIPTION
**Pull request by brcolow**
Problem:    Crash when setting v:errors to a number.
Solution:   Free the typval without assuming its type. (Yasuhiro Matsumoto)

https://github.com/vim/vim/commit/a542c680a8b42cb766e64d4ee7374ef4dacb7832

---

Not sure if the Travis failure is related?

```
Nil error
-- Output to stderr:
2016/05/11 08:57:29 [error @ open_log_file:101] 26003 - Couldn't open USR_LOG_FILE, logging to stderr! This may be caused by attempting to LOG() before initialization functions are called (e.g. init_homedir()).
2016/05/11 08:57:29 [debug @ log_server_msg:827] 26003 - [msgpack-rpc] nvim -> client(1) 2016/05/11 08:57:29 [error @ open_log_file:101] 26003 - Couldn't open USR_LOG_FILE, logging to stderr! This may be caused by attempting to LOG() before initialization functions are called (e.g. init_homedir()).
[response]     [1, 1, nil, 0]
CMake Error at /home/travis/build/neovim/neovim/cmake/RunTests.cmake:39 (message):
  Running functional tests failed with error: 1.
```

### CODE REVIEW

File: test/functional/legacy/assert_spec.lua [hunk 1]
```diff
@@ -142,4 +142,16 @@ describe('assert function:', function()
...
+  it('should allow setting to a number', function()
+      source([[
+        let save_verrors = v:errors
+        let v:['errors'] = {'foo': 3}
+        call assert_equal('yes', 'no')
+        let verrors = v:errors
+        let v:errors = save_verrors
+        call assert_equal(type([]), type(verrors))
+      ]])
+      expected_empty
```
Comment (justinmk) [inline test/functional/legacy/assert_spec.lua]: `expected_empty()`

Comment (brcolow) [inline test/functional/legacy/assert_spec.lua]: Whoops...I knew that. Thanks, fixed.

File: src/nvim/eval.c [hunk 2]
```diff
@@ -512,6 +512,8 @@ void eval_init(void)
       assert(false);
     }
   }
+  vimvars[VV_VERSION].vv_nr = VIM_VERSION_100;
```
Comment (justinmk) [inline src/nvim/eval.c]: Is there a reason not to put this at line 497, which is where it is in the original patch?

Comment (brcolow) [inline src/nvim/eval.c]: Fixed, good catch.

File: test/functional/legacy/assert_spec.lua [hunk 3]
```diff
@@ -142,4 +142,16 @@ describe('assert function:', function()
       })
     end)
   end)
+
+  it('should allow setting to a number', function()
```
Comment (jamessan) [inline test/functional/legacy/assert_spec.lua]: This is actually testing whether the old value of `v:errors` is properly freed when an `assert_*` function sets a new value, regardless of the old type of `v:errors` (which is supposed to just be a list).

Maybe reword this to `it('is reset to a list by assert functions', function()`?

Comment (brcolow) [inline test/functional/legacy/assert_spec.lua]: Fixed, thanks.

**Comment by jamessan** [comment]
Merged.  Thanks!
